libcxx/include/__tree

```diff
@@ -1277,11 +1351,28 @@ __tree<_Tp, _Compare, _Allocator>::_DetachedTreeCache::__detach_next(__node_poin
 
```

Comment:
As a follow-up, I might clean up these `__assign_multi` and `__assign_unique` methods which seem a bit like dead weight now.